backend/controllers/movieController.js

```diff
@@ -27,8 +29,44 @@ const getMovieEntry = async (req, res, next) => {
   } catch (err) { next(err); }
 };
 
+const addComment = async (req, res, next) => {
+  try {
+    const newComment = {
+      userId: req.body.userId,
+      comment: req.body.comment,
+    };
+    const imdbCode = req.params.imdb_code;
+
+    if (newComment.comment === '') throw detailedError();
+
+    const movieId = await Movie.findOneAndUpdate({
+      _id: imdbCode,
+    }, {
+      $addToSet: {
+        comments: newComment,
+      },
+    });
+
+    if (movieId === null) {
+      const newMovieInDB = new Movie({
+        _id: imdbCode,
+        comments: [
+          {
+            userId: req.body.userId,
+            comment: req.body.comment,
+          },
+        ],
+      });
+      await newMovieInDB.save();
+    }
+
+    res.sendStatus(201);
+  } catch (err) { next(err); }
```

Comment:
by the way, we can omit using try/catch block, because of the installed express-async-error package